From 569033512340d791a13c1ee2f269788c55fff63c Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Sun, 28 Jun 2020 15:19:44 +0800
Subject: [PATCH] policy/modules/system/systemd: make systemd-modules_t domain
 MLS trusted for reading from files up to its clearance

Fixes:
avc:  denied  { search } for  pid=142 comm="systemd-modules"
name="journal" dev="tmpfs" ino=10990
scontext=system_u:system_r:systemd_modules_load_t:s0-s15:c0.c1023
tcontext=system_u:object_r:syslogd_runtime_t:s15:c0.c1023 tclass=dir
permissive=0

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/systemd.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 806468109..e82a1e64a 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -739,6 +739,8 @@ modutils_read_module_objects(systemd_modules_load_t)
 
 systemd_log_parse_environment(systemd_modules_load_t)
 
+mls_file_read_to_clearance(systemd_modules_load_t)
+
 ########################################
 #
 # networkd local policy
-- 
2.17.1

